### PR TITLE
feat: add login with google

### PR DESCRIPTION
**Pull request by TMNhat1810**
contribute for: [US-2]

### CODE REVIEW

File: constants/HTTPStatusCode.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export default {
+    GOOD_REQUEST: {
+        OK: 200,
+        CREATED: 201,
+        ACCEPTED: 202,
+        NON_AUTHOR: 203,
+        NO_CONTENT: 204,
+    },
```
Comment (ces-hoangdao) [inline constants/HTTPStatusCode.js:8]: I don't think we need separate 2 kinds of requests Bad and Good, I unnecessary

**Comment by ces-hoangdao** [comment]
Please resolve sonflict

File: app/routes/auths.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import express from 'express';
+const router = express.Router();
```
Comment (ces-hoangdao) [inline app/routes/auths.js:2]: please move this line below the last import

File: app/routes/index.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import siteRouter from './sites.js';
+import authRouter from './auths.js';
 
 const route = (app) => {
   app.use('/', siteRouter);
+  app.use('/auth', authRouter);
```
Comment (ces-hoangdao) [inline app/routes/index.js:6]: Should define a constant for route to avoid magic string

File: app/routes/auths.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+router.get(
+  '/google/callback',
+  passport.authenticate('google', {
+    successRedirect: `${CLIENT_URL}`,
+    failureRedirect: '/auth/google/failure',
+  }),
+);
+
+router.get('/google/failure', authController.googleAuthFailed);
```
Comment (ces-hoangdao) [inline app/routes/auths.js:39]: should define constant for routes

**Review by ces-anduynguyen — APPROVED** [review]

**Pull request merged**